AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -162,67 +114,63 @@ - (void)setParent:(id<ASLayoutable>)parent
   }
 }
 
+
+#pragma mark - Child
+
 - (void)setChild:(id<ASLayoutable>)child
 {
-  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
+  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");;
+  ASDisplayNodeAssert(_childrenArray.count < 2, @"This layout spec does not support more than one child. Use the setChildren: or the setChild:AtIndex: API");
+  
   if (child) {
     id<ASLayoutable> finalLayoutable = [self layoutableToAddFromLayoutable:child];
     if (finalLayoutable) {
-      _children[0] = finalLayoutable;
+      _childrenArray[0] = finalLayoutable;
       [self propagateUpLayoutable:finalLayoutable];
     }
   } else {
-    _children.erase(0);
+    if (_childrenArray.count) {
+      [_childrenArray removeObjectAtIndex:0];
+    }
   }
 }
 
-- (void)setChild:(id<ASLayoutable>)child forIndex:(NSUInteger)index
+- (id<ASLayoutable>)child
 {
-  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
-  if (child) {
-    id<ASLayoutable> finalLayoutable = [self layoutableToAddFromLayoutable:child];
-    _children[index] = finalLayoutable;
-  } else {
-    _children.erase(index);
+  ASDisplayNodeAssert(_childrenArray.count < 2, @"This layout spec does not support more than one child. Use the setChildren: or the setChild:AtIndex: API");
```

Comment:
As it's only used in subclasses we should support for now until we know what the impact of the removal would have on already written code